exercises/practice/robot-simulator/.meta/example.R

```diff
@@ -0,0 +1,46 @@
+new_robot <- function(
+    coordinates = numeric(2),
+    direction = c("NORTH", "EAST", "SOUTH", "WEST")
+  ) {
+  stopifnot(is.numeric(coordinates))
+  stopifnot(length(coordinates) == 2)
+  direction <- match.arg(direction)
+  
+  structure(
+    list(
+      coordinates = as.integer(coordinates),
```

Comment:
Could your talk me through your thinking here?
It feels weird to me to define coordinates as numeric on line 2 and then convert to integer.

It's late for me here so I might be overlooking something, but I think you could just change this (line 11) to coordinates = coordinates.

An alternative would be to be stricter and expect an integer (and check with is.integer in line 5), though then one would need to pass in the coordinates as e.g. `c(0L, 0L)` and the tests would also all need to be updated to conform to this expectation.

Reply:
Above all I wanted to have default value for position as `(0, 0)`.
I started as you also suggested with `integer(2)` although I find it very unnatural for user to pass `c(1L, 1L)` so I went to numeric. 
Than I thought that such object could internally represent position in integer, helping user with the coordinated when hipothetically such robot object would be used according to requirements but accidentally passed non-integer values like `c(1.1, 3.5)`.

At the same time our tests does not force internal representation as `0 == 0L`

Reply:
Ok I figured you might have been thinking along those lines.


I agree that expecting `c(0L, 0L)` might be a bit disruptive / confusing for (some) students. I also wondered whether some students might wish to implement it with stricter integer types (which is arguably more correct) and can't do so because the tests won't pass with their numeric inputs.

But the tradeoffs you propose sound like they might be the best course of action.

Could I then suggest that we give this exercise beta status for now in the config to encourage feedback and communicate that it might still change slightly in future?

Reply:
Sure. I don't know what are the features of "beta" exercise but if you find it useful I'll just drop the change.